java/code/src/com/suse/manager/webui/controllers/StatesAPI.java

```diff
@@ -372,6 +445,144 @@ public static Object savePackages(Request request, Response response, User user)
+            }
+        } catch(TaskomaticApiException e) {
+            errors.add(e.getMessage());
+        }
+
+        if (errors.isEmpty()) {
+            return json(response, ResultJson.success());
+        }
+        return json(response, ResultJson.error(errors));
+    }
+
+    /**
+     * Updates an existing Recurring State Schedule
+     *
+     * @param request the request
+     * @param response the response
+     * @param user the user
+     * @return string containing the json response
+     */
+    public static String updateSchedule(Request request, Response response, User user) {
+        response.type("application/json");
+
+        List<String> errors = new LinkedList<>();
+
+        String scheduleId = request.params("scheduleId");
+        if (getSingleSchedule(scheduleId, user) == null) {
+            errors.add("Schedule not found.");
+        } else {
+            RecurringStateScheduleJson json = GSON.fromJson(request.body(),
+                    RecurringStateScheduleJson.class);
+            errors.addAll(saveSchedule(json, user));
+        }
+
+        if (errors.isEmpty()) {
+            return json(response, ResultJson.success());
+        }
+        return json(response, ResultJson.error(errors));
+    }
+
+    /**
+     * Saves a Recurring State Schedule
+     *
+     * @param json the RecurringStateScheduleJson Object
+     * @return String List containing errors
+     */
+    private static List<String> saveSchedule(RecurringStateScheduleJson json, User user) {
+        List<String> errors = new LinkedList<>();
+
+        String scheduleName = json.getScheduleName();
+        String type = json.getType();
+        Map<String, String> cronTimes = json.getCronTimes();
+        String cron = json.getCron();
+        String groupName = json.getGroupName();
+
+        Map<String, String> params = new HashMap<>();
+        params.put("user_id", user.getId().toString());
+        params.put("minionIds", json.getMinionIds().toString());
+        params.put("isActive", json.isActive() ? "true" : "false");
+        params.put("minionNames", json.getMinionNames().toString());
+        params.put("targetType", json.getTargetType());
+        if (groupName != null) params.put("groupName", groupName);
+        params.put("isTest", json.isTest() ? "true" : "false");
+        params.put("type", type);
+        params.putAll(cronTimes);
+
+        if (StringUtils.isEmpty(scheduleName)) {
+            errors.add("Schedule Name must be specified.");
+        }
+
+        if (errors.isEmpty()) {
+            try {
+                if (cron.isEmpty()) {
+                    RecurringEventPicker picker = RecurringEventPicker.prepopulatePicker("date", type, cronTimes, null);
+                    cron = picker.getCronEntry();
+                }
+                TaskomaticApi tapi = new TaskomaticApi();
+                tapi.scheduleSatBunch(user, scheduleName, "recurring-state-apply-bunch", cron, params);
```

Comment:
Yes, that was originally the purpose, but since I added the functionality to convert `cron` back to `cronTimes` we shouldn't need to store `cronTimes` in the database anymore.